buildSrc/src/main/kotlin/org/sonarsource/kotlin/buildsrc/tasks/CreateRuleStubsTask.kt

```diff
@@ -0,0 +1,238 @@
+package org.sonarsource.kotlin.buildsrc.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.dependencies
+import org.gradle.kotlin.dsl.provideDelegate
+import org.gradle.kotlin.dsl.repositories
+import org.sonarsource.kotlin.buildsrc.tasks.Templates.checkListFile
+import org.sonarsource.kotlin.buildsrc.tasks.Templates.checksDir
+import org.sonarsource.kotlin.buildsrc.tasks.Templates.generateCheckClass
+import org.sonarsource.kotlin.buildsrc.tasks.Templates.generateCheckFile
+import org.sonarsource.kotlin.buildsrc.tasks.Templates.generateTestClass
+import org.sonarsource.kotlin.buildsrc.tasks.Templates.samplesDir
+import org.sonarsource.kotlin.buildsrc.tasks.Templates.testsDir
+import java.nio.file.Path
+import kotlin.io.path.ExperimentalPathApi
+import kotlin.io.path.createFile
+import kotlin.io.path.notExists
+import kotlin.io.path.readLines
+import kotlin.io.path.writeLines
+import kotlin.io.path.writeText
+
+@OptIn(ExperimentalPathApi::class)
+abstract class CreateRuleStubsTask : DefaultTask() {
+
+    @get:Input
+    val ruleKey: String by project
+
+    @get:Input
+    val className: String by project
+
+    @get:Input
+    @get:Optional
+    val message: String? by project
+
+
+    @TaskAction
+    fun execute() {
+
+        val checkClassName = if (className.endsWith("Check")) {
+            className
+        } else {
+            logger.warn("""NOTE: Automatically adding "-Check" to the end of the className""")
+            "${className}Check"
+        }
+
+        logger.info("Using properties: ")
+        logger.info("  ruleKey: $ruleKey")
+        logger.info("  className: $checkClassName")
+        logger.info("  message: $message")
+
+        val finishedTasks = listOf(
+            "Create Check Class" to createCheckClass(checkClassName),
+            "Create Test Class" to createTestClass(checkClassName),
+            "Create Sample File" to createSampleFile(checkClassName),
+            "Add Rule to KotlinCheckList" to addRuleToChecksListFile(checkClassName),
+            "Download Metadata" to downloadMetadata(),
+        )
+
+        logger.info("--------- Done ---------")
+        finishedTasks.forEach { logger.info("${it.first}: ${if (it.second) "successful" else "FAILED"}") }
+    }
+
+    private fun createCheckClass(checkClassName: String): Boolean {
+        val checkFile = checksDir.resolve("${checkClassName}.kt")
+        val messageLine = message?.let { """private const val MESSAGE = "$it"""" }
+        return if (checkFile.notExists()) {
+            checkFile.createFile()
+            checkFile.writeText(generateCheckClass(ruleKey, checkClassName, messageLine))
+            true
+        } else {
+            logger.warn("WARNING: Check file '$checkFile' exists. Not creating a new one.")
+            false
```

Comment:
It's not so much a 'bad path' but to avoid overwriting potential implementations, this is defensively not touching any existing files. Since if this is the case there is a high likelihood you used the wrong class name (e.g. because you Ctrl+R the last time you used the command & forgot to change the class name), I considered this as being somewhat unexpected, although not fatal.

The boolean is basically `true` = created, `false` = not created, so indicates whether the expected action was executed (a lot like some of the other Path/File/Collection APIs work, e.g. `Path#deleteIfExists` or `MutableSet#add`). What would a status enum add here? Perhaps I am also misunderstanding your intention here.

Reply:
Ok, now I got your point here.
So if we don't want to distinguish between "File already exists, nothing to do" and "File didn't exist but sth wrong happened and creation failed" boolean is ok, if we want to, we need something more rich